imports/plugins/core/orders/client/components/OrderRefunds.js

```diff
@@ -0,0 +1,303 @@
+import React, { Fragment, useEffect, useRef, useState } from "react";
+import PropTypes from "prop-types";
+import { Mutation } from "react-apollo";
+import { Form } from "reacto-form";
+import withStyles from "@material-ui/core/styles/withStyles";
+import Card from "@material-ui/core/Card";
+import CardContent from "@material-ui/core/CardContent";
+import CardHeader from "@material-ui/core/CardHeader";
+import FormControl from "@material-ui/core/FormControl";
+import InputLabel from "@material-ui/core/InputLabel";
+import Grid from "@material-ui/core/Grid";
+import MenuItem from "@material-ui/core/MenuItem";
+import OutlinedInput from "@material-ui/core/OutlinedInput";
+import Select from "@material-ui/core/Select";
+import Typography from "@material-ui/core/Typography";
+import ErrorsBlock from "@reactioncommerce/components/ErrorsBlock/v1";
+import Field from "@reactioncommerce/components/Field/v1";
+import TextInput from "@reactioncommerce/components/TextInput/v1";
+import { i18next, Reaction } from "/client/api";
+import formatMoney from "/imports/utils/formatMoney";
+import ConfirmButton from "/imports/client/ui/components/ConfirmButton";
+import createRefundMutation from "../graphql/mutations/createRefund";
+import OrderPreviousRefunds from "./OrderPreviousRefunds";
+
+const styles = (theme) => ({
+  dividerSpacing: {
+    marginBottom: theme.spacing(4),
+    marginTop: theme.spacing(4)
+  },
+  fontWeightSemiBold: {
+    fontWeight: theme.typography.fontWeightSemiBold
+  },
+  formControl: {
+    minWidth: "250px",
+    width: "250px"
+  },
+  selectEmpty: {
+    marginTop: theme.spacing(2)
+  }
+});
+
+/**
+ * @name OrderRefunds
+ * @param {Object} props Component props
+ * @returns {React.Component} returns a React component
+ */
+function OrderRefunds(props) {
+  const hasPermission = Reaction.hasPermission(["reaction-orders", "order/fulfillment"], Reaction.getUserId(), Reaction.getShopId());
+  const { classes, order } = props;
+  const { payments } = order;
+
+  // useRef
+  const inputLabel = useRef(null);
+
+  // useState
+  const [labelWidth, setLabelWidth] = useState(0);
+  const [refundReasonSelectValues, setRefundReasonSelectValues] = useState({ reason: "" });
```

Comment:
I didn't test it, but using an object here seems unnecessary to me. 

```js
const [refundReason, setRefundReason] = useState("");

setRefundReason(event.target.value);

value={refundReason}
```

Reply:
Yes, I think you're right. It was initially taking multiple reasons, one for each payment, but then it was combined into one reason per refund.